Coding-Coach/coding-coach
# [FEATURE] Load legal content as Markdown

# Details

## Description

This PR:

* Adds a `MarkdownReader` component, which can read Markdown files from a URL prop.
* Converts Terms & Conditions, Privacy Policy and Cookies Policy legal docs into Markdown
* Updates `TermsAndConditions`, `PrivacyPolicy` and `CookiesPolicy` components to use `MarkdownReader` component.

## Steps to QA

- `npm install`
- `npm start`
- Open http://localhost:3000/
- Click links at bottom of page - "Terms & Conditions", "Cookies", "Privacy Policy"
- Verify content has loaded

## Issue

Closes #174 

## Notes

- The loaded content is unstyled. I don't know my way around tailwind and I suck at CSS anyway, so I'll leave this to a real front end dev and not just a React code monkey 😄 
- Before this PR, the email address used by the mailto link in `CookiesPolicy` was loaded from `config.contact`. Dynamically loaded text isn't possible in Markdown, so that email address is now hardcoded. I didn't see a way around this.
- Keeping these files in `src/static` probably doesn't make sense anymore. These aren't static files now, they're legit components. But this is my first PR to this project and I didn't want to just change everything without seeking input from the maintainers.
